grammar.js

```diff
@@ -75,8 +75,8 @@ module.exports = grammar({
   rules: {
     source_file: $ => seq(
       $.package_clause,
-      repeat(seq($.import_declaration, terminator)),
-      repeat(seq($._top_level_declaration, terminator))
+      repeat(seq($.import_declaration, optional(terminator))),
+      repeat(seq($._top_level_declaration, optional(terminator)))
```

Comment:
To solve the problem of parsing a file without a terminating newline or `;` I had hoped it would be simple. It's not too bad, but I ended up needing to disambiguate between `const` expressions involving `func`, and added precedence to the const, func and method declaration rules.

Reply:
:+1: Nice work figuring this out.